php-checks/src/test/resources/checks/LocalVariableShadowsClassFieldCheck.php

```diff
@@ -41,12 +53,20 @@ public function f4() {
   public function f5($param) {
     $field =& $param;                    // NOK
   }
+  
+  public function f6() {
+    $field = foo();                      // NOK
+    callback(function() use ($field)  {  // OK
+    });
+  }
+
 }
 
 class B {
   public $field;
 
-  public function __construct($field) {  // OK (Constructor)
+  public function __construct() {
+    $field = foo();                      // OK (Constructor)
     $this->field = $field;
   }
 
```

Comment:
@pynicolas code on line 26: it's not clear why it's ok - because `$field` is parameter or because function is static. If second, then we don't test local variables inside static function, only parameters